prov/verbs/src/verbs_domain.c

```diff
@@ -325,8 +325,10 @@ vrb_domain(struct fid_fabric *fabric, struct fi_info *info,
 	struct vrb_fabric *fab =
 		 container_of(fabric, struct vrb_fabric,
 			      util_fabric.fabric_fid);
+	ofi_mutex_lock(&vrb_info_mutex);
 	struct fi_info *fi = vrb_get_verbs_info(vrb_util_prov.info,
 						info->domain_attr->name);
+	ofi_mutex_unlock(&vrb_info_mutex);
```

Comment:
vrb_get_verbs_info() is only called from here.  Can you move the function into this file (or integrate the functionality inline), only pass in the domain name, and take the mutex within the function?